src/hooks/useSafeTokenAllocation.ts

```diff
@@ -39,25 +99,63 @@ const fetchAllocation = async (chainId: string, safeAddress: string): Promise<Ve
   }
 }
 
-const useSafeTokenAllocation = (): BigNumber | undefined => {
-  const [allocation, setAllocation] = useState<BigNumber>()
+const fetchTokenBalance = async (chainId: string, safeAddress: string): Promise<string> => {
+  try {
+    const web3ReadOnly = getWeb3ReadOnly()
+    const safeTokenAddress = getSafeTokenAddress(chainId)
+    if (!safeTokenAddress || !web3ReadOnly) return '0'
+
+    const tokenBalance = await web3ReadOnly?.call({
```

Comment:
I decided to fetch this on-chain instead of our redux store, because the remainder of this data (tokens claimed, etc) is also fetched on-chain so the data will always be in sync like this.

Reply:
I noticed we do 6 additional RPC calls on load now. Its something we should probably keep track of and maybe optimize e.g. batching `eth_call`